### ISSUE TITLE
LID Task is deceased

### ISSUE THREAD

**Issue by xiayq1** [yeyupiaoling/Whisper-Finetune#110]
I find after FT whisper, the ASR improved, but the language identity detection(LID) decreased. Why?

**Comment by xiayq1** [comment]
First, thanks for sharing your Whisper finetuning code—it's been really helpful!

While reviewing the implementation, I noticed a minor bug related to label processing. Here's what I found:

I check transformer in WhisperEncoder(WhisperPreTrainedModel) it will using :
shift_tokens_right, to shift the label and add sot token at beginning:

`if labels is not None:
            if labels.shape[1] > self.max_target_positions:
                raise ValueError(
                    f"Labels' sequence length {labels.shape[1]} cannot exceed the maximum allowed length of {self.max_target_positions} tokens."
                )
            if decoder_input_ids is None and decoder_inputs_embeds is None:
                decoder_input_ids = shift_tokens_right(
                    labels, self.config.pad_token_id, self.config.decoder_start_token_id
                )`

But whisper processor, the label will automatically become  [sot,  lang_id, task_tok, notimestep, ....].

`    def __getitem__(self, idx):
        try:
            # 从数据列表里面获取音频数据、采样率和文本
            sample, sample_rate, transcript, language = self._get_list_data(idx=idx)
            # 可以为单独数据设置语言
            self.processor.tokenizer.set_prefix_tokens(language=language if language is not None else self.language)
            if len(transcript) > 0:
                # 加载带有时间戳的文本
                if self.timestamps:
                    data = self._load_timestamps_transcript(transcript=transcript)
                    # 从输入音频数组中计算log-Mel输入特征
                    data["input_features"] = self.processor(audio=sample, sampling_rate=self.sample_rate).input_features
                else:
                    # 获取log-Mel特征和标签ID
                    data = self.processor(audio=sample, sampling_rate=self.sample_rate, text=transcript)
`

So in the end,  which means the labels is [sot, sot, lang_id, task_tok, ....]. So the proformance in LID task will decease.

I solve the problem by adding:

`
data['labels'] = data['labels'][1:]
`

Just wanted to flag this in case it’s helpful—you might want to double-check the label processing logic in your Whisper finetuning code. The fix I mentioned above (trimming the duplicate sot token) could improve performance, especially for language-specific tasks.

If you get a chance to review it, this might also benefit others who are using your code for Whisper finetuning. Thanks again for sharing your work!

**Comment by xiayq1** [comment]
I find in this repo: https://huggingface.co/blog/fine-tune-whisper.

It also using the 
`
if (labels[:, 0] == self.decoder_start_token_id).all().cpu().item():
            labels = labels[:, 1:]
`
